lib/rouge/lexers/clean.rb

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Clean < RegexLexer
+      title "Clean"
+      desc "The Clean programming language (clean.cs.ru.nl)"
+
+      tag 'clean'
+      filenames '*.dcl', '*.icl'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          if otherwise
+          let in
+          with where
+          case of
+          infix infixl infixr
+          class instance
+          generic derive
+          special
+          implementation definition system module
+          from import qualified as
+          dynamic
+          code inline foreign export ccall stdcall
+        )
+      end
+
+      state :common_literals do
+        rule %r/'(?:[^'\\]|\\(?:x[0-9a-fA-F]+|\d+|.))'/, Str::Char
+
+        rule %r/[+~-]?\d+\.\d+(?:E[+-]?\d+)?\b/, Num::Float
+        rule %r/[+~-]?\d+E[+-]?\d+\b/, Num::Float
+        rule %r/[+~-]?\d+/, Num::Integer
+
+        rule %r/"/, Str::Double, :string
+      end
+
+      state :basic do
+        rule %r/\s+/m, Text::Whitespace
+
+        rule %r/\/\/\*.*/, Comment::Doc
```

Comment:
The example in the document says that multiline doc comments begin with `/**` but there is no rule that matches that pattern. Instead, it's tokenised as a multiline comment. Is that intentional?

Reply:
No, that was an oversight. Thanks for catching it. The `:comment` and `:comment_doc` states now contain some duplicate code, is there a neat way around this?

Reply:
It's sometimes possible to avoid duplication by having rules be mixed in but that won't help here where you want to use different tokens. We have some lexers where people use some of Ruby's metaprogramming tricks to avoid duplicating code but I'm of the view that the obfuscation this results in outweighs any benefit from removing duplicate code .

In short, I don't think a little duplication is probelmatic, especially when it makes maintenance easier (which I think it does in this case).

Reply:
If the above is satisfactory from your side, the code looks ready to merge :)

Reply:
Yes, for me it's fine :) thanks for your time and the in-depth review!